src/main/java/org/example/springbootintro/model/Book.java

```diff
@@ -9,10 +9,12 @@
 import java.math.BigDecimal;
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
 
 @Entity
 @Setter
 @Getter
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
```

Comment:
You should add restriction for use soft-de;ete.
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
@SQLRestriction(value = "is_deleted = false")
```